src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -32,193 +32,668 @@
+            fieldMappingSetName = MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME;
+        }
+
+        Map<String, Data_Import_Field_Mapping__mdt> fieldMappingsByFieldMappingSetDeveloperName =
+            helper.getFieldMappingsByFieldMappingSetDeveloperName(fieldMappingSetName);
+
+        for (DataImportFieldMapping fieldMapping : dataImportFieldMappings) {
+
+            String fieldMappingKey = fieldMapping.targetObjectAPIName +
+                '.' + fieldMapping.dataImportObjectName +
+                '.' + fieldMapping.sourceFieldAPIName;
+
+            Data_Import_Object_Mapping__mdt existingDefaultObjectMapping =
+                objectMappingsByDeveloperName.get(fieldMapping.dataImportObjectName);
+            Data_Import_Field_Mapping__mdt existingDefaultFieldMapping =
+                fieldMappingsByFieldMappingSetDeveloperName.get(fieldMappingKey);
+
+            if (existingDefaultFieldMapping == null) {
+                // Is a new field mapping
+                fieldMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+                fieldMapping.setObjectMapping(existingDefaultObjectMapping);
+                // Setting name to null will force field mapping to generate a unique name
+                fieldMapping.setRecordName(null, false);
+
+                queuedMetadataTypesForDeploy.add(fieldMapping.getMetadataRecord());
+
+            } else if (existingDefaultFieldMapping != null) {
+
+                if (customFieldMappingSetExists) {
+                    // Migrated Custom Field Mapping Set exists
+                    DataImportFieldMapping overwrittenMapping =
+                        new DataImportFieldMapping(
+                            existingDefaultFieldMapping,
+                            MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+
+                    overwrittenMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+                    overwrittenMapping.setObjectMapping(existingDefaultObjectMapping);
+                    overwrittenMapping.setTargetField(fieldMapping.targetFieldAPIName);
+
+                    queuedMetadataTypesForDeploy.add(overwrittenMapping.getMetadataRecord());
+                } else {
+                    // Check if we have a conflict and update the target field based on the help text mapping
+                    if (fieldMapping.hasBadDefaultValues(existingDefaultFieldMapping)) {
+                        fieldMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+                        fieldMapping.setObjectMapping(existingDefaultObjectMapping);
+                        fieldMapping.setTargetField(fieldMapping.targetFieldAPIName);
+                    } else {
+                        fieldMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+                        fieldMapping.setObjectMapping(existingDefaultObjectMapping);
+                    }
```

Comment:
`fieldMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
fieldMapping.setObjectMapping(existingDefaultObjectMapping);`
Can be moved above the if block and only performed once since they are doing the same thing in each block.